File: packages/core/src/schemas/comment.ts
import { z } from "zod";
import { CardIdSchema } from "./card.js";

export const CommentIdSchema = z.string().brand<"CommentId">();

export const CommentAuthor = z.enum(["user", "system", "ai"]);

export const CommentSchema = z.object({
  id: CommentIdSchema,
  card_id: CardIdSchema,
  author: CommentAuthor,
  content: z.string().min(1),
  execution_id: z.string().nullable().default(null),
  created_at: z.string(),
});

export const CreateCommentSchema = z.object({
  author: CommentAuthor.default("user"),
  content: z.string().min(1).max(50_000),
  execution_id: z.string().nullable().optional().default(null),
});

File: rust/crates/core/src/types/comment.rs
use serde::{Deserialize, Serialize};
use ts_rs::TS;

#[derive(Debug, Clone, Serialize, Deserialize, TS, PartialEq)]
#[ts(export, export_to = "../../../../packages/dashboard/src/types/generated/")]
#[serde(rename_all = "lowercase")]
pub enum CommentAuthor {
    User,
    System,
    Ai,
}

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[ts(export, export_to = "../../../../packages/dashboard/src/types/generated/")]
pub struct Comment {
    pub id: String,
    pub card_id: String,
    pub author: CommentAuthor,
    pub content: String,
    pub execution_id: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Clone, Deserialize, TS)]
#[ts(export, export_to = "../../../../packages/dashboard/src/types/generated/")]
pub struct CreateComment {
    pub author: CommentAuthor,
    pub content: String,
    pub execution_id: Option<String>,
}
